src/config/mod.rs

```diff
@@ -3974,6 +3983,39 @@ impl TikvConfig {
         }
     }
 
+    pub fn optional_default_cfg_adjust_with(
+        &mut self,
+        last_cfg: &Option<Self>,
+    ) -> Result<(), Box<dyn Error>> {
+        let kv_data_exists = self.kv_data_exists()?;
+        let is_titan_dir_empty = self.titan_data_exists()?;
```

Comment:
```suggestion
        let is_titan_dir_empty = !self.titan_data_exists()?;
```

Reply:
In fact, I don't think we need to check titan dir empty logic. Instead, we can just read it from last_cfg for titan enabling flag as well.

Reply:
For the rare case, last config may be deleted manually. So it's better to keep checking titan dir